presentation/src/main/java/com/moez/QKSMS/feature/backup/RestoreBackupService.kt

```diff
@@ -81,30 +81,30 @@ class RestoreBackupService : Service() {
         startForeground(NOTIFICATION_ID, notification.build())
 
         backupRepo.getRestoreProgress()
-                .sample(200, TimeUnit.MILLISECONDS, true)
-                .subscribeOn(Schedulers.io())
-                .subscribe { progress ->
-                    when (progress) {
-                        is BackupRepository.Progress.Idle -> stop()
-
-                        is BackupRepository.Progress.Running -> notification
-                                .setProgress(progress.max, progress.count, progress.indeterminate)
-                                .setContentText(progress.getLabel(this))
-                                .let { notificationManager.notify(NOTIFICATION_ID, it.build()) }
-
-                        else -> notification
-                                .setProgress(0, 0, progress.indeterminate)
-                                .setContentText(progress.getLabel(this))
-                                .let { notificationManager.notify(NOTIFICATION_ID, it.build()) }
-                    }
+            .sample(200, TimeUnit.MILLISECONDS, true)
+            .subscribeOn(Schedulers.io())
+            .subscribe { progress ->
+                when (progress) {
+                    is BackupRepository.Progress.Idle -> stop()
+
+                    is BackupRepository.Progress.Running -> notification
+                        .setProgress(progress.max, progress.count, progress.indeterminate)
+                        .setContentText(progress.getLabel(this))
+                        .let { notificationManager.notify(NOTIFICATION_ID, it.build()) }
+
+                    else -> notification
+                        .setProgress(0, 0, progress.indeterminate)
+                        .setContentText(progress.getLabel(this))
+                        .let { notificationManager.notify(NOTIFICATION_ID, it.build()) }
                 }
+            }
 
         // Start the restore
         Observable.just(intent)
-                .map { Uri.parse(it.getStringExtra(EXTRA_FILE_URI)) }
-                .map(backupRepo::performRestore)
-                .subscribeOn(Schedulers.io())
-                .subscribe({}, Timber::w)
+            .map { Uri.parse(it.getStringExtra("${baseContext.packageName}.$EXTRA_FILE_URI}")) }
```

Comment:
I think there was a typo, this may have been causing the null pointer exception.
```suggestion
            .map { Uri.parse(it.getStringExtra("${baseContext.packageName}.$EXTRA_FILE_URI")) }
```

Reply:
i've just pushed a new commit to this pr which removes the extraneous brace